tmk_core/protocol/usb_descriptor.c

```diff
@@ -1061,17 +1065,58 @@ const USB_Descriptor_String_t PROGMEM ProductString = {
     .UnicodeString              = USBSTR(PRODUCT)
 };
 
-#if defined(SERIAL_NUMBER)
+// clang-format on
+
+#if SERIAL_NUMBER_USE_HARDWARE_ID == TRUE
+
+#    if defined(__AVR__)
+#        error Dynamically setting the serial number on AVR is unsupported as LUFA requires the string to be in PROGMEM.
+#    endif // defined(__AVR__)
+
+#    ifndef SERIAL_NUMBER_LENGTH
+#        define SERIAL_NUMBER_LENGTH 32
```

Comment:
Could we do something with `sizeof(hardware_id_t)` just incase that structure gets larger for some reason in the future? I think it would be....

```suggestion
#        define SERIAL_NUMBER_LENGTH (sizeof(hardware_id_t) * 2)
```

Reply:
Yeah I had that initially but then clang was complaining that it was multiplying two sizeof's.
Seems fine now, not sure.